### ISSUE TITLE
Example for connecting to auth emulator results in runtime exception.

### ISSUE THREAD

**Issue by Turburlar** [firebase/firebase-js-sdk#6744]
I am developing using firebase 9.13.0 library in a node.js 14.17.1 environment.

I have followed the example code to connect to a running emulator found in the documentation [here...](https://firebase.google.com/docs/emulator-suite/connect_auth#web-version-9)

My resulting code is specifically as follows...

`const firebaseAuth = require('firebase/auth');
const auth = firebaseAuth.getAuth();
firebaseAuth.connectAuthEmulator(auth, 'http://localhost:9099');`

This code results in the error...

`Firebase: Need to provide options, when not being deployed to hosting via source. (app/no-options).`

My emulator is confirmed to be running and accessible using the firebase-admin-sdk in a node.js environment as well as the flutter firebase client libraries.

**Comment by hsubox76** [comment]
How are you initializing the Firebase app? (What does your `initializeApp()` call look like?)

**Comment by Turburlar** [comment]
> How are you initializing the Firebase app? (What does your `initializeApp()` call look like?)

I am not calling initializeApp. No where in this documentation on 'how to connect to emulator' does it indicate that you need to, or how you are supposed to call initializeApp....especially with a demo project (with no actual cloud resources created).

Dealing with firebase documentation is like swimming in a pool full or vinegar and razor blades. lol

**Comment by hsubox76** [comment]
That's not supposed to be the full code of the app, it's a snippet showing how to do the connection part specifically. Emulators are for testing a Firebase app you have already made ([how to create a Firebase web app](https://firebase.google.com/docs/web/setup#add-sdks-initialize)) so you have to make a regular app first. Then you add the code from the connection snippets to connect to an emulator instead of connecting to the real backend.

I think the [intro to the emulators documentation](https://firebase.google.com/docs/emulator-suite#before_you_begin) explains that you should first look at the Get Started docs and also suggested downloading a quickstart app such as Friendly Eats.

Let me know if you think that's not clear or prominent enough and if you have any ideas on how to make it more clear.

**Comment by Turburlar** [comment]
In the very document we are both referring to, it indicates that a demo project has no actual firebase resources created...

`
A demo Firebase project has no real Firebase configuration and no live resources. These projects are usually accessed via codelabs or other tutorials.Project IDs for demo projects have the demo- prefix. | When working with demo Firebase projects, your apps and code interact with emulators only. If your app attempts to interact with a resource for which an emulator isn't running, that code will fail.`

Show me where in this document it has a single initializeApp call, let alone an example of initializeApp with a demo project.

You would think the document instructing how to use the emulator would give at least one complete minimal working example...

**Comment by hsubox76** [comment]
That's correct, you don't have to create a project in the console. "Resources" refer to backend resources. You still have to write some client-side code to set up a client-side app that will be using the emulators. That's why it suggests getting these projects from "codelabs or other tutorials".

I can refer this to our documentation team but I'll close the issue since there doesn't seem to be a bug in the code.
